### PR TITLE
Dedicated Material classes for Standard, URP and HDRP

### PR DESCRIPTION
**Pull request by Roland09**
Created dedicated material converters and ball material creators for Standard, URP and HDRP graphic pipelines

### CODE REVIEW

File: VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs [hunk 1]
```diff
@@ -3,144 +3,75 @@
...
-				debug?.AppendLine("Metallic set to 1.");
-			}
+		private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
 
-			// roughness / glossiness
-			unityMaterial.SetFloat(Glossiness, vpxMaterial.Roughness);
+		/// <summary>
+		/// Lazy instantiation of the material converter
+		/// </summary>
+		private static readonly Lazy<IMaterialConverter> lazyMaterialConverterInstance = new Lazy<IMaterialConverter>(() => CreateMaterialConverter());
```
Comment (rbxnk) [inline VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs:21]: Lazy isn't something I come across much, is there a good reason to use this over simply having an IMaterialConver member and in the property just doing something like `Foo => _foo ?? _foo = new bar()`?

File: VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs [hunk 2]
```diff
@@ -3,144 +3,75 @@
...
-			// blend mode
-			ApplyBlendMode(unityMaterial, vpxMaterial.MapBlendMode);
-			if (vpxMaterial.MapBlendMode == BlendMode.Translucent) {
-				col.a = Mathf.Min(1, Mathf.Max(0, vpxMaterial.Opacity));
-				unityMaterial.SetColor(Color, col);
-			}
+		/// <summary>
+		/// Singleton Material Converter
+		/// </summary>
+		public static IMaterialConverter MaterialConverter { get { return lazyMaterialConverterInstance.Value; } }
```
Comment (rbxnk) [inline VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs:26]: We've been preferring the => shorthand whenever possible

**Comment by rbxnk** [comment]
This all looks awesome. I do wonder though why we need the separation of the ball materials. I know all that predates this change, but I wonder if we could just route them through the table's mat/factory?

File: VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs (hunk 1 above)
Comment (Roland09) [reply VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs]: Thanks, removed it.

File: VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs (hunk 2 above)
Comment (Roland09) [reply VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs]: Thanks, I updated the code.

File: VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs [hunk 3]
```diff
@@ -16,15 +16,9 @@ public static class MaterialExtensions
...
+		/// Material Converter instance for the current graphics pipeline
 		/// </summary>
-		private static readonly Lazy<IMaterialConverter> lazyMaterialConverterInstance = new Lazy<IMaterialConverter>(() => CreateMaterialConverter());
-
-		/// <summary>
-		/// Singleton Material Converter
-		/// </summary>
-		public static IMaterialConverter MaterialConverter { get { return lazyMaterialConverterInstance.Value; } }
-
+		public static IMaterialConverter MaterialConverter => CreateMaterialConverter();
```
Comment (rbxnk) [inline VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs:21]: Sorry didn't mean the idea of a singleton was bad, was just curious about the benefits of Lazy over a simple singleton one liner. I think we probably would prefer not to instantiate these very call

**Comment by freezy** [comment]
I don't mind having the ball creation separate. It allows for more flexibility, and since the ball isn't part of the table, we don't need to keep backwards compatibility.

File: VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs (hunk 3 above)
Comment (freezy) [reply VisualPinball.Unity/VisualPinball.Unity/Extensions/MaterialExtensions.cs:21]: Reading [this](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization) it simply seems a thread-safe way of doing the same thing. If it's part of .NET Standard and supported by Unity I don't mind using it.

**Pull request merged**